STATISTICS.md

```diff
@@ -0,0 +1,523 @@
+type | string | `"producer"` | Instance type (producer or consumer)
+ts | int | 12345678912345 | librdkafka's internal monotonic clock (micro seconds)
+time | int | | Wall clock time in seconds since the epoch
+replyq | int gauge | | Number of ops waiting in queue for application to serve with rd_kafka_poll()
+msg_cnt | int gauge | | Current number of messages in instance queues
+msg_size | int gauge | | Current total size of messages in instance queues
+msg_max | int | | Threshold: maximum number of messages allowed
+msg_size_max | int | | Threshold: maximum total size of messages allowed
+tx | int | | Total number of requests sent
+txbytes | int | | Total number of bytes sent
+rx | int | | Total number of responses received
+rxbytes | int | | Total number of bytes received
+txmsgs | int | | Total number of messages transmitted (produced)
+txmsg_bytes | int | | Total number of bytes transmitted for txmsgs
+rxmsgs | int | | Total number of messages consumed, not including ignored messages (due to offset, etc).
+rxmsg_bytes | int | | Total number of bytes received for rxmsgs
+simple_cnt | int gauge | | Internal tracking of legacy vs new consumer API state
+metadata_cache_cnt | int gauge | | Number of topics in the metadata cache.
+brokers | object | | Dict of brokers, key is broker name, value is object. See **brokers** below
+topics | object | | Dict of topics, key is topic name, value is object. See **topics** below
+cgrp | object | | Consumer group metrics. See **cgrp** below
+
+## brokers
+
+Field | Type | Example | Description
+----- | ---- | ------- | -----------
+name | string | `"example.com:9092/13"` | Broker hostname, port and broker id
+nodeid | int | 13 | Broker id (-1 for bootstraps)
+state | string | `"UP"` | Broker state (INIT, DOWN, CONNECT, AUTH, APIVERSION_QUERY, AUTH_HANDSHAKE, UP, UPDATE)
+stateage | int gauge | | Time since last broker state change (microseconds)
+outbuf_cnt | int gauge | | Number of requests awaiting transmission to broker
+outbuf_msg_cnt | int gauge | | Number of messages in outbuf_cnt
+waitresp_cnt | int gauge | | Number of requests in-flight to broker awaiting response
+waitresp_msg_cnt | int gauge | | Number of messages in waitresp_cnt
+tx | int | | Total number of requests sent
+txbytes | int | | Total number of bytes sent
+txerrs | int | | Total number of transmission errors
+txretries | int | | Total number of request retries
+req_timeouts | int | | Total number of requests timed out
+rx | int | | Total number of responses received
+rxbytes | int | | Total number of bytes received
+rxerrs | int | | Total number of receive errors
+rxcorriderrs | int | | Total number of unmatched correlation ids in response (typically for timed out requests)
+rxpartial | int | | Total number of partial messagesets received
+zbuf_grow | int | | Total number of decompression buffer size increases
+buf_grow | int | | Total number of buffer size increases
```

Comment:
maybe a bit more detail. what buffers? presumably does not include decompression buffers?